src/codeu/chat/client/simplegui/ChatGuiFX.java

```diff
@@ -262,26 +280,121 @@ private void signUpButtonClicked(ActionEvent e) {
+    // index handles possible duplicate titles.
+    private ConversationSummary lookupByTitle(String title, int index) {
 
-        if (name != null && name.length() > 0) {
-            // TODO: users need to be added to the server first? So I can add the conversation to that user???
-            //clientContext.conversation.startConversation(name, clientContext.user.getCurrent().id);
-            convoList.addAll(name);
+      int localIndex = 0;
+      for (final ConversationSummary cs : clientContext.conversation.getConversationSummaries()) {
+        if ((localIndex >= index) && cs.title.equals(title)) {
+          return cs;
         }
+        localIndex++;
+      }
+      return null;
     }
 
-    private void selectConversation(MouseEvent e) {
-        // set the conversation title
-        chatTitle.setText(conversations.getSelectionModel().getSelectedItem());
+    // External agent calls this to trigger an update of this panel's contents.
+    public void update(ConversationSummary owningConversation) {
+
+        final User u = (owningConversation == null) ?
+            null :
+            clientContext.user.lookup(owningConversation.owner);
+
+        chatTitle.setText(owningConversation.title);
+
+        getAllMessages(owningConversation);
+    }
+
+    // Populate ListModel - updates display objects.
+    private void getAllConversations(ListView<String> convDisplayList) {
+
+      clientContext.conversation.updateAllConversations(false);
+      convDisplayList.getItems().clear();
+
+      for (final ConversationSummary conv : clientContext.conversation.getConversationSummaries()) {
+        convoList.addAll(conv.title);
+      }
+    }
+
+
+    private void sendMessage(ActionEvent e) {
+        if (!clientContext.user.hasCurrent()) {
+            // if the user is not signed in
+            Alert alert = new Alert(AlertType.ERROR);
+            alert.setHeaderText("You're not signed in!");
+            alert.showAndWait();
+
+        } else if (!clientContext.conversation.hasCurrent()) {
+            // if the user did not select or add a conversation
+            Alert alert = new Alert(AlertType.ERROR);
+            alert.setHeaderText("Select or add a conversation on the right!");
+            alert.showAndWait();
+
+        } else {
+            String messageText = input.getText();
+            messageList.addAll(input.getText());
+            if (messageText != null && messageText.length() > 0) {
+                // add message to current conversation
+                clientContext.message.addMessage(clientContext.user.getCurrent().id,
+                    clientContext.conversation.getCurrentId(), messageText);
+
+                // populate the list
+                ChatGuiFX.this.getAllMessages(clientContext.conversation.getCurrent());
+            }
+        }
+    }
+
+    // method to populate the list of messages
+    private void getAllMessages(ConversationSummary conversation) {
```

Comment:
I'd rename this function. It's not actually returning anything, so calling it `getSomething()` is a little misleading. Maybe something like `fetchMessagesAndUpdateGui()`? Don't be afraid of long function names.